code/modules/client/preferences_toggles.dm

```diff
@@ -454,3 +454,12 @@ GLOBAL_LIST_INIT(ghost_orbits, list(GHOST_ORBIT_CIRCLE,GHOST_ORBIT_TRIANGLE,GHOS
 		return
 	prefs.asaycolor = initial(prefs.asaycolor)
 	prefs.save_preferences()
+
+/client/verb/toggle_darkmode()
+	set name = "Toggle darkmode"
+	set category = "Preferences"
+	set desc = "Change your chat theme"
+	prefs.toggles ^= DARKMODE
+	prefs.save_preferences()
+	to_chat(usr, "Dark theme (YOU MUST RECONNECT TO SEE THIS CHANGE) [(usr.client.prefs.toggles & DARKMODE) ? "Enabled" : "Disabled"]")

```

Comment:
What's up with this? Can't you just pile on the winsets?

Reply:
Yeah it's more so goonchat reloads as I didn't want to break it by loading it again.

Reply:
I guess I could delete their browseroutput and make a new one?

Reply:
Rather than sending one of light/dark CSS as the single css file to goonchat, send both and add some way to tell goonchat to switch which is active.

Reply:
Again, still requires goonchat to reload no?

Reply:
should be possible to do without a reload

Reply:
But wouldn't the client still have the wrong assets? Or would the new ones override the old.

Reply:
Send both assets always, instruct goonchat to change which one is being used

Reply:
Ohhhhhhh I'll give that a shot.

Reply:
Actually that still won't help. If they go from dark -> white they'll still have the wingets applied and must relog anyway.

Reply:
Second function which `winset`s everything the other way

Reply:
I know 🤔 
That just seems like a lot of extra spaghetti for quite a small issue /shrug

Reply:
Maybe personal preference, but forcing users to relog for no good reason is a huge red flag that a feature is not well-thought-ought IMO. In the past, changing UI style only applied on relog, so it basically just gave off the appearance that that preference didn't do anything. #38089 involved refactoring to fix it but now the feature actually works rather than being confusing.

Reply:
Mmm, but surely we can fix it with some explicit messaging about needing to log out and back in.


I'm sympathetic to the concern of having to have two winset functions that need to be kept up to date with style changes.

Reply:
My main concern is that, yeah. As there's no way to automate this every ui update means going through two sets of winsets which is horrible...

Reply:
SpaceManiac is right here.

You could just make the different modes be defined by a pair of datums and have winsets able to apply the whole thing from those

Reply:
That still requires manually replacing every window though...

If you can find someone willing to maintain that I'll do it.